nixos/modules/virtualisation/virtualisation-options.nix

```
@@ -0,0 +1,61 @@
+# This modules declares shared options for virtual machines,
+# containers and anything else in `virtualisation`.
+#
+# This is useful to declare e.g. defaults for
+# `virtualisation.diskSize` once, while building multiple
+# different image formats of a NixOS configuration.
+#
+# Additional options can be migrated over time from
+# `modules/virtualisation/qemu-vm.nix` and others.
+# Please keep defaults and descriptions here generic
+# and independent of i.e. hypervisor-specific notes
+# and defaults where.
+# Those can be added in the consuming modules where needed.
+# needed.
+let
+  _file = ./virtualisation-options.nix;
+  key = _file;
+in
+{
+  diskSize =
+    { lib, config, ... }:
+    let
+      t = lib.types;
+    in
+    {
+      inherit _file key;
```

This should really just be a module file called `disk-size.nix`; no surprises, no workarounds.

```suggestion
{ lib, config, ... }:
    let
      t = lib.types;
    in
    {
```

(continuation of [thread](https://github.com/NixOS/nixpkgs/pull/341058/files#r1761806176))

---

> Additional options can be migrated over time from

Doesn't really belong in code. It'd be better to open a an issue for "project status"-like info.

---

The idea was to keep generic options separate from the platform/provider specific images. Doesn't need to be one file, did disk-size-option.nix for now. Could be a sub directory once there's more options

---

Got it. It seems that you haven't pushed this change yet.

---

Oh indeed! Should be pushed now